repose-aggregator/components/filters/cors-filter/src/main/scala/org/openrepose/filters/cors/CorsFilter.scala

```diff
@@ -178,23 +204,71 @@ class CorsFilter @Inject()(configurationService: ConfigurationService)
         Nil
     })
   }
+
+  def getHostUri(request: HttpServletRequestWrapper): URI = {
+    val maybeForwardedHost = request.getSplittableHeaderScala(CommonHttpHeader.X_FORWARDED_HOST).headOption
+    val maybeForwardedHostUri = Try(maybeForwardedHost
+      .map(forwardedHost => new URIBuilder(s"${request.getScheme}://$forwardedHost"))
+      .map(uri => uri.setPort(normalizePort(uri.getPort, uri.getScheme)).setHost(normalizeUriHost(uri.getHost)).build()))
+    lazy val hostUri = new URIBuilder().setScheme(request.getScheme).setHost(normalizeHost(request.getServerName))
```

Comment:
On second thought, perhaps I shouldn't. In this case, we really aren't concerned with where the request is going, but rather with where it is now. If the two are kept in-sync, then we lose that distinction (due to the nature of being a reverse proxy).